### ISSUE TITLE
How to use object literals in array with no-object-literal-type-assertion?

### ISSUE THREAD

**Issue by CSchulz** [palantir/tslint#2923]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.3.4
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
@NgModule({
  providers: [
    <FactoryProvider> {
      deps: [LOG_ROOT_FACTORY],
      provide: LOG_FACTORY,
      useFactory: createLogger,
    },
  ],
})
```

with `tslint.json` configuration:

```json
"no-object-literal-type-assertion": true
```

#### Behavior
It complains about the `<FactoryProvider>`.

It would be nice to have an example how to use it in such cases. Should every provider hold as const?

**Comment by ajafff** [comment]
Can you elaborate what the actual problem is?

Just remove the type assertion and everything should be fine. If there's a compile error without the assertion, there's probably something wrong with your code.

I know you use this pattern to get code completion inside the object literal. If that doesn't work without the assertion, that's better reported in the typescript repo.

**Comment by CSchulz** [comment]
It is just used for better code completion and I am just wondering how to declare it there.

**Comment by ajafff** [comment]
Maybe this is related and eventually solves your issues: https://github.com/Microsoft/TypeScript/issues/16457
